src/nexmark/queries/q2.rs

```diff
@@ -0,0 +1,130 @@
+use super::NexmarkStream;
+use crate::{nexmark::model::Event, operator::FilterMap, Circuit, OrdZSet, Stream};
+
+/// Selection
+///
+/// Find bids with specific auction ids and show their bid price.
+/// See https://github.com/nexmark/nexmark/blob/v0.2.0/nexmark-flink/src/main/resources/queries/q2.sql
+const AUCTION_ID_MODULO: u64 = 123;
+
+pub fn q2(input: NexmarkStream) -> Stream<Circuit<()>, OrdZSet<(u64, usize), isize>> {
+    input
+        .filter(|event| match event {
```

Comment:
This can be more efficiently implemented by replacing `filter().map()` with `flat_map()`, using `Option<(auction, price)>` as output iterator type.

Reply:
Done, thanks.

Reply:
`.filter_map()` would be a better translation

Reply:
We no longer have `filter_map`.  It's a special case of `flat_map` using `Option<>` as output type.